docs/source/development.rst

```diff
@@ -271,3 +271,118 @@ Also, agent releases are numbered in line with the deployment branch name.
+
+
+Testing a new deployment
+------------------------
+
+After deploying Beamer to a chain, it is critical to check the new deployment
+to make sure that not only the contracts work correctly, but also that the
+agent software works with the new deployment. To that end, there are two kinds
+of checks that need to be performed: L1 invalidations and challenges.
+
+L1 invalidations
+~~~~~~~~~~~~~~~~
+
+Checking whether L1 invalidations work correctly is done as follows:.
+
+#. Let ``X1``, ``X2``,... ``Xn`` be IDs of all chains that are supported by Beamer.
+   Also assume that one of those, ``C``, is the ID of the chain you just deployed to.
+
+   First, arrange for invalidations to be sent from ``C`` to all chains::
+
+     beamer check initiate-l1-invalidations --output invalidations.json C X1 X2 X3 ... Xn
+
+   Note that this command will issue multiple invalidations, by default one for
+   each chain pair ``(C, Xi)``.
+
+   Then, arrange for invalidations to be sent from each chain ``Xi`` to ``C``::
+
+     for x in X1 X2 ... Xn; do
+       beamer check initiate-l1-invalidations --output invalidations.json $x C
+     done
+
+   This is to make sure that the other direction also works.
+   Note that we use the same file, ``invalidations.json`` to collect all invalidations,
+   regardless of the directions.
+
+#. Verify invalidations::
+
+     beamer check verify-l1-invalidations invalidations.json
+
+   This will verify all invalidations from ``invalidations.json``. Due to the
+   fact that different chains may have different finalization periods, it is
+   possible that some invalidations succeed while others fail. This is expected
+   and you should be able to re-run the command later to check again. Once the
+   finalization period is over for every invalidation, the command should
+   successfully verify all of them. Typically, after a week all invalidations
+   should be finalized.
+
+
+Challenges
+~~~~~~~~~~
+
+Checking whether challenges are working correctly follows the same pattern
+as L1 invalidation checks, however, it is a bit more involved because an
+agent needs to be configured and running on all involved chains.
+
+The procedure is as follows:
+
+#. Let ``X1``, ``X2``,... ``Xn`` be IDs of all chains that are supported by Beamer.
+   Also assume that one of those, ``C``, is the ID of the chain you just deployed to.
+
+   First, make sure to have a deployed test token on each chain.
+
+.. note:: The test token `must` be an ERC20 token.
+
+   In general, it's easiest to just deploy ``MintableToken`` when deploying Beamer
+   (see ``--deploy-mintable-token`` option of the ``beamer deploy`` command).
+
+   From this point on, we will assume that ``MintableToken`` (``TST``) will be used.
+
+#. Configure and start a single agent instance that monitors all ``X1`` ... ``Xn`` chains.
+   The test token address needs to be configured properly on each chain in the agent
```

Comment:
There is the information missing that the token also needs to be configured in the contracts (transfer limit > 0 and ideally ethInToken = 0)

Reply:
Good point, added a separate step describing that. :+1: